### PR TITLE
feat(server): Amazon ElastiCache Memcached MCP Server

### PR DESCRIPTION
**Pull request by seaofawareness**
Fixes

## Summary
Create the first version of official Amazon ElastiCache Memcached MCP Server

### Changes
* Basic CRUD operations
* Multi-key operations
* Counter operations
* Server management functions
* SSL/TLS connection support
* Error handling, Edge cases
> Please provide a summary of what's being changed

### User experience

User can use any LLM client to do natural language to Valkey data type operations

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented

Is this a breaking change? (Y/N)
N

**RFC issue number**: 297

Checklist:

* [x] Migration process documented
* [x] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

### CODE REVIEW

**Comment by eyesfivemserver** [comment]
I'm impressed by the modularity of this codebase. Makes it very maintainable.

File: src/memcached-mcp-server/awslabs/memcached_mcp_server/main.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    """Run the MCP server with CLI argument support."""
+    parser = argparse.ArgumentParser(
+        description='An AWS Labs Model Context Protocol (MCP) server for Amazon ElastiCache Memcached'
+    )
+    parser.add_argument('--sse', action='store_true', help='Use SSE transport')
+    parser.add_argument('--port', type=int, default=8888, help='Port to run the server on')
+
+    args = parser.parse_args()
+
+    logger.trace('A trace message.')
```
Comment (krokoko) [inline src/memcached-mcp-server/awslabs/memcached_mcp_server/main.py]: these logger calls should be removed

Comment (seaofawareness) [reply src/memcached-mcp-server/awslabs/memcached_mcp_server/main.py]: Removed

**Review by awslabs-mcp — APPROVED** [review]
## PR Review: Amazon ElastiCache Memcached MCP Server

I've completed a review of this PR which creates the first version of the official Amazon ElastiCache Memcached MCP Server. This server provides an MCP interface to interact with Amazon ElastiCache Memcached through a secure and reliable connection.

### Summary of Changes

- Adds a new MCP server component for Amazon ElastiCache Memcached
- Implements basic CRUD operations, multi-key operations, and counter operations
- Provides secure connection handling with SSL/TLS support
- Includes error handling and edge case management
- Well-documented and thoroughly tested with good test coverage

### Architecture and Implementation

The MCP server follows the standard project structure used across other MCP servers in the monorepo. It provides natural language interfaces to interact with Memcached caches, enabling AI agents to efficiently manage and search cached data.

The implementation provides robust connection management with:
- Configurable connection settings and timeouts
- Automatic connection retrying with customizable parameters
- Connection pooling for improved performance

### Documentation and Testing

- The server is well-documented with clear usage instructions
- Comprehensive tests cover the core functionality
- README provides clear setup instructions and usage examples

### Integration with Existing Code

The server integrates well with the rest of the MCP ecosystem by:
- Following the established project structure and conventions
- Providing consistent configuration patterns
- Properly updating documentation in various places (README.md, docs folder)

All CI checks are passing, indicating that the code follows the project standards and quality requirements.

### Final Assessment

This is a valuable addition to the MCP server ecosystem, enabling interaction with Amazon ElastiCache Memcached instances through natural language. The code is well-structured, thoroughly tested, and adheres to project standards.

Run ID: 15058363807

**Review by awslabs-mcp — APPROVED** [review]
## PR Review: Amazon ElastiCache Memcached MCP Server

I've completed a review of this PR which creates the first version of the official Amazon ElastiCache Memcached MCP Server. This MCP server provides an interface to interact with Amazon ElastiCache Memcached clusters through natural language instructions.

### Summary of Changes

- Adds a new MCP server component for Amazon ElastiCache Memcached
- Implements core functionality including:
  - Basic CRUD operations for cache keys
  - Multi-key operations for efficient batch processing
  - Counter operations for atomic increments/decrements
  - Server management functions
  - SSL/TLS connection support
- Comprehensive test coverage (96.61% patch coverage)
- Well-documented interfaces and usage examples

### Architecture and Implementation

The implementation follows the project's standard MCP server structure and design guidelines:
- Well-organized into appropriate modules
- Clean separation of concerns (connection handling, cache operations)
- Proper environment variable configuration
- Secure connection handling with SSL/TLS support

### Documentation and Testing

- The server includes comprehensive documentation in the README.md
- Test coverage is excellent at 96.61% (with only 9 lines missing coverage)
- Code quality checks and security scans have all passed

### Integration with Existing Code

The PR properly integrates with the MCP ecosystem:
- Updates main README.md to include the new server
- Updates documentation in the docs folder
- Follows project structure and coding conventions
- Configuration follows established patterns across other MCP servers

### Final Assessment

This is a valuable addition to the MCP server ecosystem that enables natural language interactions with Amazon ElastiCache Memcached. The code is well-structured, thoroughly tested, and adheres to project standards.

Run ID: 15061048953

**Review by krokoko — APPROVED** [review]
